### DISCUSSION TITLE
Is it possible to animate ListView items?

### DISCUSSION THREAD

**Discussion by Vadoola** [slint-ui/slint#4932] [Q&A] [answered]
I am trying to animate the items inside a ListView as the items get added or removed (in this instance specifically a filter model).

A couple simple video examples of the ideas I'm trying to implement:

- Using Swift Ui: items fade in and slide up as they appear: https://www.youtube.com/watch?v=tgXDZTv-tMM
- Using Flutter, Items sliding in from the right: https://www.youtube.com/watch?v=s_VYapaJWeE

The idea that I'm specifically trying to implement is that when the filter is applied the items no being shown slide out, and when the filter is removed the items slide back in.

I have a rough example I created to test the idea that can be found at https://github.com/Vadoola/anim_list_items.

Pressing the Toggle button filters in/out the "Text 2". In the slint code there are states and in/out animations to change the background color and opacity of the Rectangle. I've tried a handful of things and none of them seem to work. The items just appear and disappear instantly, with no animation.

Is what I'm trying to achieve currently possible with Slint?

**Comment by Vadoola** [comment]
Ah, I guess what I essentially would be looking for are enter and leave animations which appear to not yet be supported:  https://github.com/slint-ui/slint/issues/3258

**Reply by ogoffart** [reply]
Yes, that's the issue, but this is not yet implemented.
Maybe as of now there might be way to start an animation in "init"?

**Reply by Vadoola** [reply, marked as answer]
> Yes, that's the issue, but this is not yet implemented. Maybe as of now there might be way to start an animation in "init"?

For future reference to anyone who comes across this. I wasn't able to get anything working with "init" (maybe's it's possible but I didn't figure it out).

I did however get something working. It's a bit hacky, and I don't know if it will scale well, but it can be seen in [this demo repo](https://github.com/Vadoola/anim_list_items).
Essentially it uses a timer in the rust code to delay the filter_model.reset() when sliding out, and then uses an extra property tied to the position, and delays setting that with a timer after calling the filter_model.reset() on sliding in.

It would certainly be a lot cleaner, and less error prone to have enter and leave animations, but until those are added, for my purposes I might be able to get this to work in the meantime.
